src/ast/passes/semantic_analyser.cpp

```diff
@@ -2819,7 +2819,15 @@ void SemanticAnalyser::visit(AssignMapStatement &assignment)
 
   const std::string &map_ident = assignment.map->ident;
 
-  if (type.IsRecordTy() && map_val_[map_ident].IsRecordTy()) {
+  if (type.IsCastableMapTy() && map_val_[map_ident].IsIntTy()) {
+    if (auto *call_expr = dynamic_cast<Call *>(assignment.expr)) {
```

Comment:
nit: to remove the unused variable warning
```suggestion
    if (dynamic_cast<Call *>(assignment.expr)) {
```